### DISCUSSION TITLE
DataFetcherExceptionHandler.onException() is deprecated, but how to replace it with handleException()?

### DISCUSSION THREAD

**Discussion by albertogdd** [Netflix/dgs-framework#743] [Q&A] [answered]
Based on the [DGS docs regarding custom exceptions mapping](https://netflix.github.io/dgs/error-handling/#mapping-custom-exceptions), its recommended to use `onException()` method from `DataFetcherExceptionHandler` interface. But this method is deprecated by graphql and recommends using `handleException()` instead.

Maybe the question doesn't belong here but, how can we only use `handleException()` while, as being an interface, both methods must be implemented? We have some warnings regarding deprecation and would like to get rid of them.

I believe this is a recent graphql library change, some info I found about it
https://githubmemory.com/repo/graphql-java/graphql-java/issues/2545

Having a look at DGS's `DefaultDataFetcherExceptionHandler` you only use the deprecated `onException()` (I believe, not a pro with Kotlin)

**Comment by srinivasankavitha** [comment, marked as answer]
We'll need to update the framework to handle this. We did already switch to graphql-java-17, so just a matter of adding the new method in the framework. In the meantime, you can write your own exception handler as well without using the DefaultDataFetcherExceptionHandler and wire that in using the same mechanism.

**Reply by albertogdd** [reply]
Thank you Kavitha
